src/gui_common/AddWindowReorderingSupportToSiblings.cs

```diff
@@ -0,0 +1,263 @@
+﻿using System.Collections.Generic;
+using Godot;
+using Godot.Collections;
+using static CustomWindow;
+
+public class AddWindowReorderingSupportToSiblings : Control
+{
+    [Export]
+    public Array<NodePath> WindowReorderingPaths = new();
+
+    /// <summary>
+    ///   Finds first window reordering node to connect to
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     Ignored when window reordering paths are not empty
+    ///   </para>
+    /// </remarks>
+    [Export]
+    public int AutomaticWindowReorderingDepth = 0;
+
+    [Export]
+    public bool ReorderWindows = true;
```

Comment:
It can work as a middleman (to pass connections to its ancestors). Theoretically it can be used to automatically connect windows that are too deep in the hierarchy. I believe that this is an edge case where it will usually be better to set a manual path for those so if you think that this is not a good enough reason to have it then I will remove it.

Reply:
I think it should be removed. I saw a depth control variable as well, so reaching up a bunch would work by increasing that as well, probably?

Reply:
It exists for performance reasons and to prevent it from accidentally connecting to something one doesn't want it to connect to. Letting it to try to connect to every single ancestor sounds like something that should be prevented but otherwise performance shouldn't be an issue, I think? 10 sounds like too much to me because it's rarely needed if ever. What about 5?

It's possible but I can't think of anything to improve it. For now I will try to write the comments.

Reply:
I think 10 is a nice round number and the performance difference shouldn't be too different from 5, I think.
And also the case anyway should be that each level of windows would have their reordering support, because missing reordering support is a sign of programming bug once this is done. I guarantee that someone at some point will notice and comment about a bug where just one window in the game doesn't go to the top of others when clicked.

Reply:
Okay. What about a connection between two levels of windows (So from a reordering support node to its reordering support ancestor)? The same should apply to those too, right?

Reply:
Well my initial thought would be that the sibling reorder would intelligently determine which node to move even for a really deep window. But also the design which you seem to want to go for where you have multiple levels of reorder sibling nodes, is probably good. In this case as you'd usually not have the higher level sibling reorder node I think here maybe 5 would be a better search depth as most of the time the search would fail and hit the limit.